shopify/monthly-subscription-template/api/models/shopifyShop/actions/subscribe.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  applyParams,
+  preventCrossShopDataAccess,
+  save,
+  ActionOptions,
+} from "gadget-server";
+import { trialCalculations } from "../helpers";
+
+export const run: ActionRun = async ({
+  params,
+  record,
+  logger,
+  api,
+  connections,
+  currentAppUrl,
+}) => {
+  applyParams(params, record);
+  await preventCrossShopDataAccess(params, record);
+
+  if (params.planId) {
+    const planMatch = await api.plan.maybeFindOne(params.planId, {
+      select: {
+        name: true,
+        monthlyPrice: true,
+        currency: true,
+        trialDays: true,
+      },
+    });
+
+    if (planMatch) {
+      const today = new Date();
+
+      // Check for trial availability
+      const { usedTrialMinutes, availableTrialDays } = trialCalculations(
+        record.usedTrialMinutes,
+        record.usedTrialMinutesUpdatedAt,
+        today,
+        planMatch.trialDays
+      );
+
+      if (!planMatch.monthlyPrice) {
+        throw new Error("ZERO COST PLAN - The price of a plan cannot be zero");
+      }
+
+      logger.info(
+        { env: process.env.NODE_ENV === "production" ? "false" : "true" },
+        "HERE"
+      );
+
+      /**
+       * Create subscription record in Shopify
+       * Shopify requires that the price of a subscription be non-zero. This template does not currently support free plans
+       */
+      const result = await connections.shopify.current?.graphql(
+        `mutation ($name: String!, $lineItems: [AppSubscriptionLineItemInput!]!, $returnUrl: URL!) {
+          appSubscriptionCreate(name: $name, returnUrl: $returnUrl, lineItems: $lineItems) {
```

Comment:
Missing test in mutation definition - this will fix the issue with test payments not working
```suggestion
        `mutation ($name: String!, $lineItems: [AppSubscriptionLineItemInput!]!, $returnUrl: URL!, $test: Boolean!) {
          appSubscriptionCreate(name: $name, returnUrl: $returnUrl, lineItems: $lineItems, test: $test) {
```